spec/performance/reek/smell_detectors/control_parameter_spec.rb

```diff
@@ -0,0 +1,19 @@
+require_relative '../../../spec_helper'
+require_lib 'reek/smell_detectors/control_parameter'
+require_lib 'reek/examiner'
+require 'stackprof'
+
+RSpec.describe Reek::SmellDetectors::ControlParameter do
+  let(:source) { SAMPLES_DIR.join('smelly_source').join('ruby.rb') }
+
+  it 'runs on a complicated source file in less than 10 seconds' do
+    examiner = Reek::Examiner.new source, filter_by_smells: ['ControlParameter']
+    # This is the actual test
+    # expect(examiner.smells.size).to perform_under(10).sec
```

Comment:
@mvz stumbled across [this gem](https://github.com/piotrmurach/rspec-benchmark) - seems worth a shot to me. Also wdyt about those kind of performance specs in general?

Reply:
Yes, definitely worth a try. I have no experience with this kind of spec but it looks very useful.